stdlib/src/array.rs

```diff
@@ -230,12 +230,20 @@ mod array {
                     }
                 }
 
-                fn index(&self, obj: PyObjectRef, vm: &VirtualMachine) -> PyResult<usize> {
+                fn index(
+                    &self,
+                    obj: PyObjectRef,
+                    start: usize,
+                    stop: usize,
+                    vm: &VirtualMachine
+                ) -> PyResult<usize> {
                     match self {
                         $(ArrayContentType::$n(v) => {
                             if let Ok(val) = <$t>::try_into_from_object(vm, obj) {
-                                if let Some(pos) = v.iter().position(|&a| a == val) {
-                                    return Ok(pos);
+                                for (index, element) in v.iter().enumerate().take(stop as usize).skip(start as usize) {
+                                    if element == &val {
+                                        return Ok(index);
+                                    }
```

Comment:
Can we use `position` again instead of manual comparision?

```suggestion
                                if let Some(pos) = v.iter().take(stop as _).skip(start as _).position(|&elem| elem == val) {
                                    return Ok(pos + start);
```

Reply:
I think your suggestion is very good
I'll try as you suggest. Thank you !